frontend/src/components/user/UserTable.vue

```diff
@@ -2,45 +2,42 @@
 import { ref } from 'vue';
 
 import { Button, Column, DataTable } from '@/lib/primevue';
-import { useAuthZStore } from '@/store';
-import { NavigationPermission } from '@/store/authzStore';
 import { AccessRequestStatus } from '@/utils/enums/application';
 
 import type { Ref } from 'vue';
 import type { UserAccessRequest } from '@/types';
 
 // Props
-const { usersAndAccessRequest, revocation = false } = defineProps<{
-  usersAndAccessRequest: Array<UserAccessRequest>;
-  revocation?: boolean;
+const { usersAndAccessRequests, requestTable = false } = defineProps<{
+  usersAndAccessRequests: Array<UserAccessRequest>;
+  requestTable?: boolean;
 }>();
 
 // Constants
 const DEFAULT_SORT_ORDER = 1;
 const DEFAULT_SORT_FIELD = 'fullName';
 
 // Emits
-const emit = defineEmits([
-  'userTable:processRequest',
-  'userTable:denyRevocation',
-  'userTable:delete',
-  'userTable:manage',
-  'userTable:revoke'
-]);
-
-// Store
-const authzStore = useAuthZStore();
+const emit = defineEmits(['userTable:approveRequest', 'userTable:denyRequest', 'userTable:manage', 'userTable:revoke']);
 
 // State
 const selection: Ref<UserAccessRequest | undefined> = ref(undefined);
+
+function getStatusColour(data: UserAccessRequest) {
+  let colour = 'black';
+  if (data.accessRequest && data.user.status !== AccessRequestStatus.APPROVED) {
+    colour = data.accessRequest.grant ? '#036' : '#D8292F';
+  }
+  return colour;
+}
 </script>
```

Comment:
We should not be defining hex values to be inserted into a style prop. Use Vue's ability to dynamically set CSS classes. See the <span> element below for more info.